posts: announce rustup 1.28.0 (#1494)
rust-lang/blog.rust-lang.org: posts/2025-03-02-Rustup-1.28.0.md

---
layout: post
title: "Announcing Rustup 1.28.0"
author: The Rustup Team
---

The rustup team is happy to announce the release of rustup version 1.28.0.
[Rustup][install] is the recommended tool to install [Rust][rust], a programming language that is empowering everyone to build reliable and efficient software.

> If you have a previous version of rustup installed, getting rustup 1.28.0 is as easy as stopping any programs which may be using Rustup (e.g. closing your IDE) and running:

Reviewer: Suggest we do "how to upgrade" after "what's new".

Author: @djc This is pre-existing, and I believe it's there to be coherent with Rust's release announcements such as https://blog.rust-lang.org/2025/01/09/Rust-1.84.0.html.

    advantage of the platform's certificate store on platforms that support it.
- When creating proxy links, rustup will now try symlinks first and fall back to hardlinks,
  as opposed to trying hardlinks first.
- A new `RUSTUP_LOG` environment variable can be used to control tracing-based logging in
  rustup binaries. See the [dev guide](https://rust-lang.github.io/rustup/dev-guide/tracing.html) for more details.

Finally, there are some notable changes to our [official website][install] as well:

- The overall design of the website has been updated to better align with the Rust Project's branding.
- It is now possible to download the prebuilt `rustup-init.sh` installer for the `aarch64-pc-windows-msvc` host platform via https://win.rustup.rs/aarch64.

Further details are available in the [changelog]!

Rustup's documentation is also available in [the rustup book][book].

[other installation methods]: https://rust-lang.github.io/rustup/installation/other.html
[changelog]: https://github.com/rust-lang/rustup/blob/stable/CHANGELOG.md
[book]: https://rust-lang.github.io/rustup/

## Thanks

Thanks again to all the contributors who made rustup 1.28.0 possible!

> - Alovchin91 (Alovchin91)

Reviewer: I'm inclined to drop the list of contributors, maybe referring to the changelog instead.

Author: This corresponds to https://blog.rust-lang.org/2025/01/09/Rust-1.84.0.html's "Contributors to 1.84.0" section, but I have no particular preference WRT keeping this section as-is or delegate to the changelog.

Reviewer: That doesn't inline the list of contributors either, though, right?

<img width="819" alt="Screenshot 2025-03-02 at 15 31 39" src="https://github.com/user-attachments/assets/d3260132-c6b6-4cfd-a204-845097105e48" />

Author: @djc The full changelog, I believe, is there in case someone wants to find out what have happened in this release cycle, whereas https://thanks.rust-lang.org gives a list of _contributors_ to express our gratitude instead of contributions, and I believe this is the main difference here. We should been mentioned in that website as well since a long time ago but alas... As I said I have no particular preference regarding what should be done right here & right now.

Reviewer 2: rustup has a website so I wonder if we could add a thanks section somewhere there?

Reviewer 3: Just a fly-by. I opened this: <https://github.com/rust-lang/thanks/issues/77>.

But yeah maybe rustup can publish it on its own website.

---
layout: post
title: "Announcing Rustup 1.28.0"
author: The Rustup Team
---

The rustup team is happy to announce the release of rustup version 1.28.0.
[Rustup][install] is the recommended tool to install [Rust][rust], a programming language that is empowering everyone to build reliable and efficient software.

## What's new in rustup 1.28.0

This new release of rustup has been a long time in the making and comes with substantial changes.

Before digging into the details, it is worth mentioning that [Chris Denton](https://github.com/chrisdenton) has joined the team.

> Chris has a lot of experience contributing to Windows-related parts of the Rust Project, expertise which was sorely lacking in existing team members, so we're happy to have him on board to help address Windows-specific issues.

Reviewer 4: ```suggestion
Chris has a lot of experience contributing to Windows-related parts of the Rust Project -- expertise we were previously lacking -- so we're happy to have him on board to help address Windows-specific issues.
```

It reads a bit better and more positively to write that aside in the second person than in the passive voice.
